commit 2393ba1c86ef70a953ca94b34bcab02f97f2db2c
Author: cheyang <cheyang@163.com>
Date:   Fri May 25 01:55:31 2018 +0800

    refactor fluentd-elasticsearch (#5696)
    
    * refactor fluentd-elasticsearch
    
    * remove trailing spaces
    
    * Update Chart.yaml

diff --git a/incubator/fluentd-elasticsearch/Chart.yaml b/incubator/fluentd-elasticsearch/Chart.yaml
index 314997a81..614f850eb 100644
--- a/incubator/fluentd-elasticsearch/Chart.yaml
+++ b/incubator/fluentd-elasticsearch/Chart.yaml
@@ -1,5 +1,5 @@
 name: fluentd-elasticsearch
-version: 0.3.0
+version: 0.4.0
 appVersion: 2.0.4
 home: https://www.fluentd.org/
 description: A Fluentd Helm chart for Kubernetes with Elasticsearch output
diff --git a/incubator/fluentd-elasticsearch/README.md b/incubator/fluentd-elasticsearch/README.md
index 6342c5b1f..ee113613d 100644
--- a/incubator/fluentd-elasticsearch/README.md
+++ b/incubator/fluentd-elasticsearch/README.md
@@ -50,9 +50,9 @@ The following table lists the configurable parameters of the Fluentd elasticsear
 | `elasticsearch.buffer_queue_limit` | Elasticsearch buffer queue limit           | `8`                                                        |
 | `extraVolumeMounts`                | Mount an extra volume, required to mount ssl certificates when elasticsearch has tls enabled |          |
 | `extraVolume`                      | Extra volume                               |                                                            | 
-| `image`                            | Image                                      | `gcr.io/google-containers/fluentd-elasticsearch`           |
-| `imageTag`                         | Image tag                                  | `v2.0.4                                                    |
-| `imagePullPolicy`                  | Image pull policy                          | `Always` if `imageTag` is `imagePullPolicy`                |
+| `image.repository`                 | Image                                      | `gcr.io/google-containers/fluentd-elasticsearch`               |
+| `image.tag`                        | Image tag                                  | `v2.0.4                                                    |
+| `image.pullPolicy`                 | Image pull policy                          | `Always` if `image.tag` is `imagePullPolicy`                |
 | `rbac.create`                      | RBAC                                       | `true`                                                     |
 | `resources.limits.cpu`             | CPU limit                                  | `100m`                                                     |
 | `resources.limits.memory`          | Memory limit                               | `500Mi`                                                    |
diff --git a/incubator/fluentd-elasticsearch/templates/daemonset.yaml b/incubator/fluentd-elasticsearch/templates/daemonset.yaml
index 9aa3e9782..7f7ad4c2b 100644
--- a/incubator/fluentd-elasticsearch/templates/daemonset.yaml
+++ b/incubator/fluentd-elasticsearch/templates/daemonset.yaml
@@ -4,7 +4,7 @@ metadata:
   name: {{ template "fluentd-elasticsearch.fullname" . }}
   labels:
     app: {{ template "fluentd-elasticsearch.fullname" . }}
-    version: {{ .Values.imageTag }}
+    version: {{ .Values.image.tag }}
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
@@ -14,18 +14,18 @@ spec:
   selector:
     matchLabels:
       app: {{ template "fluentd-elasticsearch.fullname" . }}
-      version: {{ .Values.imageTag }}
+      version: {{ .Values.image.tag }}
       chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
       heritage: "{{ .Release.Service }}"
   template:
     metadata:
       labels:
         app: {{ template "fluentd-elasticsearch.fullname" . }}
-        version: {{ .Values.imageTag }}
+        version: {{ .Values.image.tag }}
         chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
         heritage: "{{ .Release.Service }}"
         kubernetes.io/cluster-service: "true"
-        version: {{ .Values.imageTag }}
+        version: {{ .Values.image.tag }}
       # This annotation ensures that fluentd does not get evicted if the node
       # supports critical pod annotation based priority scheme.
       # Note that this does not guarantee admission on the nodes (#40573).
@@ -39,7 +39,7 @@ spec:
       serviceAccountName: {{ template "fluentd-elasticsearch.fullname" . }}
       containers:
       - name: {{ template "fluentd-elasticsearch.fullname" . }}
-        image:  "{{ .Values.image }}:{{ .Values.imageTag }}"
+        image:  "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.imagePullPolicy | quote }}
         env:
         - name: FLUENTD_ARGS
diff --git a/incubator/fluentd-elasticsearch/values.yaml b/incubator/fluentd-elasticsearch/values.yaml
index dcd0dfbff..678ebeb49 100644
--- a/incubator/fluentd-elasticsearch/values.yaml
+++ b/incubator/fluentd-elasticsearch/values.yaml
@@ -1,10 +1,10 @@
-image: gcr.io/google-containers/fluentd-elasticsearch
-imageTag: v2.0.4
-
+image:
+  repository: gcr.io/google-containers/fluentd-elasticsearch
 ## Specify an imagePullPolicy (Required)
 ## It's recommended to change this to 'Always' if the image tag is 'latest'
 ## ref: http://kubernetes.io/docs/user-guide/images/#updating-images
-imagePullPolicy: IfNotPresent
+  tag: v2.0.4
+  pullPolicy: IfNotPresent
 
 ## Configure resource requests and limits
 ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
